### PR TITLE
Fix nulls in join

### PR DESCRIPTION
**Pull request by TomyMeren**
# Description

It takes into account when we have null values on the left, right or both sides of the Join.

SparqQL always tries to get a value from a join hand instead of a null. This approach includes a possible combination of the fields

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala [hunk 1]
```diff
@@ -144,17 +144,23 @@ class ConstructSpec
       }
       """
 
-      Compiler
-        .compile(df, None, query, config)
-        .right
-        .get
-        .collect()
-        .toSet shouldEqual Set(
+      val result = Compiler.compile(df, None, query, config)
```
Comment (TomyMeren) [inline modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala:147]: @patofp can you help me check this result? I have a strange result in Jena

File: modules/engine/src/test/scala/com/gsk/kg/engine/compiler/GraphSpec.scala [hunk 2]
```diff
@@ -1829,8 +1829,18 @@ class GraphSpec
         val result = Compiler.compile(df, None, query, config)
 
         result shouldBe a[Right[_, _]]
-        result.right.get.collect.length shouldEqual 1
+        result.right.get.collect.length shouldEqual 3
         result.right.get.collect.toSet shouldEqual Set(
+          Row(
```
Comment (TomyMeren) [inline modules/engine/src/test/scala/com/gsk/kg/engine/compiler/GraphSpec.scala:1834]: this is the test with which we detected the problem

File: modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala:147]: I've checked the result on Jena and the result matches the previous version:

```
Set(
        Row(
          "_:test",
          "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>",
          "<http://id.gsk.com/dm/1.0/Document>"
        ),
        Row(
          "_:test",
          "<http://id.gsk.com/dm/1.0/docSource>",
          "\"source\""
        )
```

**Review by patofp — CHANGES_REQUESTED** [review]

File: modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala (hunk 1 above)
Comment (TomyMeren) [reply modules/engine/src/test/scala/com/gsk/kg/engine/compiler/ConstructSpec.scala:147]: Resolved! Thank you @patofp.
It turns out that when a row have all the columns of the join condition null, they never cross.
I solved with a filter to remove those rows before the join.

File: modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala [hunk 3]
```diff
@@ -370,6 +374,72 @@ final case class Multiset(
       .drop(leftGraphCol)
       .drop(rightGraphCol)
   }
+
+  def innerJoinEmptyValues(
```
Comment (patofp) [inline modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala:378]: I suggest to document this method, what it does and what are the steps. What do you think?

Comment (TomyMeren) [reply modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala:378]: Yes, sounds good!

Comment (TomyMeren) [reply modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala:419]: Done!

File: modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala [hunk 4]
```diff
@@ -370,6 +374,113 @@ final case class Multiset(
       .drop(leftGraphCol)
       .drop(rightGraphCol)
   }
+
+  /** Check if the df's have nulls values in any of his rows. if don't have, we
```
Comment (patofp) [inline modules/engine/src/main/scala/com/gsk/kg/engine/Multiset.scala:378]: Nice!

**Review by patofp — APPROVED** [review]
LGTM! GJ!

**Pull request merged**